### PR TITLE
Add test to check if TXT public area is marked reserved in e820 map

### PR DESCRIPTION
**Pull request by ChriMarMe**
Signed-off-by: Christopher Meis <christopher.meis@9elements.com>

### CODE REVIEW

File: pkg/test/memory.go [hunk 1]
```diff
@@ -249,6 +260,18 @@ func TXTRegisterSpaceValid(txtAPI hwapi.APIInterfaces) (bool, error, error) {
 	return true, nil, nil
 }
 
+// TXTPublicReservedInE820 checks if TXTPublic area is marked reserved in e820 map
+func TXTPublicReservedInE820(txtAPI hwapi.APIInterfaces) (bool, error, error) {
+	res, err := txtAPI.IsReservedInE820(uint64(tools.TxtPublicSpace), uint64(tools.TxtPublicSpace+0x10000))
+	if err != nil {
+		return false, nil, err
+	}
+	if res {
```
Comment (zaolin) [inline pkg/test/memory.go:269]: Check result here

File: pkg/test/memory.go [hunk 2]
```diff
@@ -249,6 +260,18 @@ func TXTRegisterSpaceValid(txtAPI hwapi.APIInterfaces) (bool, error, error) {
 	return true, nil, nil
 }
 
+// TXTPublicReservedInE820 checks if TXTPublic area is marked reserved in e820 map
+func TXTPublicReservedInE820(txtAPI hwapi.APIInterfaces) (bool, error, error) {
+	res, err := txtAPI.IsReservedInE820(uint64(tools.TxtPublicSpace), uint64(tools.TxtPublicSpace+0x10000))
```
Comment (zaolin) [inline pkg/test/memory.go:265]: Is it always 0x10000? Is it an offset?

Comment (ChriMarMe) [reply pkg/test/memory.go:265]: Document Number: 558294, Revision: 2.0, page 48

File: pkg/test/memory.go [hunk 3]
```diff
@@ -17,6 +17,16 @@ var (
 		SpecificiationTitle:     IntelTXTSpecificationTitle,
 		SpecificationDocumentID: IntelTXTSpecificationDocumentID,
 	}
+	testtxtpublicisreserved = Test{
+		Name:                    "TXT public area reserved in e820",
+		Required:                true,
+		function:                TXTPublicReservedInE820,
+		dependencies:            []*Test{&testtxtmemoryrangevalid},
+		Status:                  Implemented,
+		SpecificationChapter:    "5.5.4 TPM Decode Area",
```
Comment (PatrickRudolph) [inline pkg/test/memory.go:26]: wrong chapter, should be "5.5.3 Intel TXT Public Space"

**Pull request merged**